clients/tests/unit/render/Fields/Mocks/MockFileRenderer.test.js

```diff
@@ -0,0 +1,78 @@
+import {shallow,mount} from 'enzyme';
+import EnzymeAdapter from '../../../createEnzymeAdapter'
+import {fileFieldConfigs} from "./fileFieldConfigs";
+import {MockFileFieldRenderer} from "./MockFileFieldRenderer";
+import React from 'react';
+
+describe( 'DOM testing file components', () => {
+
+	it( 'Can change value', () => {
+		const field = fileFieldConfigs.required_single_allow_png;
+		const component = mount(
+			<MockFileFieldRenderer
+				field={field}
+
+			/>
+		);
+		component.instance().onChange( 'seven' );
+		expect(component.state('value')).toBe('seven')
+	});
+
+	it( 'Can setIsInvalid', () => {
+		const field = fileFieldConfigs.required_single_allow_png;
+		const component = mount(
+			<MockFileFieldRenderer
+				field={field}
+
+			/>
+		);
+		component.instance().setIsInvalid( true );
+		expect(component.state('isInvalid')).toBe(true);
+	});
+
+
+	it( 'Can set disabled', () => {
+		const field = fileFieldConfigs.required_single_allow_png;
+		const component = mount(
+			<MockFileFieldRenderer
+				field={field}
+			/>
+		);
+		component.instance().setIsInvalid( true );
+		expect(component.state('isInvalid')).toBe(true);
+	});
+
+
+	it( 'Can set message', () => {
+		const field = fileFieldConfigs.required_single_allow_png;
+		const component = mount(
+			<MockFileFieldRenderer
+				field={field}
+
+			/>
+		);
+		const message = {error: true, message: 'Fail' };
+		component.instance().setMessage(message );
+		expect(component.state('message')).toBe(message)
+	});
+
+});
+
+describe( 'DOM testing file components', () => {
+
+
+	it( 'If multiple upload option is not enabled, can not add a second file', () => {
```

Comment:
@New0 Here is an example test. Doesn't even require a file blob. I'm just checking based on number of values.

We need to use file blobs for testing the preview.

For the onDrop/onReject, we should use file blob to test that blobs get passed to array properly. We do not need to test that dropzone knows how to reject/accept files.

We need to test that when a message that indicates an error is passed to  the field that it shows correctly.